src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -323,9 +323,14 @@ address ArchiveBuilder::reserve_buffer() {
   _shared_rs = rs;
 
   _buffer_bottom = buffer_bottom;
-  _current_dump_region = &_rw_region;
-  _num_dump_regions_used = 1;
-  _current_dump_region->init(&_shared_rs, &_shared_vs);
+
+  if (CDSConfig::is_dumping_static_archive()) {
+    _current_dump_region = &_pz_region;
+    _current_dump_region->init(&_shared_rs, &_shared_vs);
```

Comment:
Second line in 'if' and 'else' seems to be identical ?

Reply:
fixed